### PR TITLE
Connect to API on startup

### PR DESCRIPTION
**Pull request by olorin**

### CODE REVIEW

File: src/TatooineCli.hs [hunk 1]
```diff
@@ -93,21 +93,22 @@ doDownload (DownloadEnv dir c) = do
...
 doUpload (UploadEnv dir retention c) = do
+  -- Connect to the API before doing anything else, so we fail fast in
+  -- the case of a configuration error.
+  creds <- bimapEitherT TatooineCredentialLoadError id $ obtainCredentialsForUpload (apiKey c) (apiEndpoint c)
   prepareDir dir
   now   <- liftIO $ getCurrentTime
   (incoming, processing, bads)  <- processDir dir (NoChangeAfter $ twoMinutesBefore now)
   liftIO $ mapM_ warnBadFile bads
-  uploaded <- upload' processing
+  uploaded <- upload' creds processing
```
Comment (nhibberd) [inline src/TatooineCli.hs]: Id be tempted to clean this up further, not sure if the `upload'` is useful anymore. At least I don't see the need to explicitly pass `creds`.

Maybe something like `mapM (const $ uploadReady dir Sydney creds) processing`, not overly fussed though

**Comment by nhibberd** [comment]
Consider above but :+1:

**Comment by charleso** [comment]
:+1:

File: src/TatooineCli.hs (hunk 1 above)
Comment (olorin) [inline src/TatooineCli.hs]: Good point, fixed.

**Pull request merged**